cloudsync/smartsync.py

```diff
@@ -351,7 +359,8 @@ def smart_listdir_path(self, local_path):
             lent = local_dir_ents.get(name)
             if not rent or not rent[LOCAL].path or self.translate(LOCAL, rent[REMOTE].path) == rent[LOCAL].path:
                 yield_val = self._ent_to_smartinfo(rent, lent, local.join(local_path, name))
-                yield yield_val
+                if yield_val:
```

Comment:
If the filtering is moved here then the return value for `_ent_to_smartinfo` isn't `Optional`